### PR TITLE
[INFRA] Rename build_system to cmake

### PR DESCRIPTION
**Pull request by eseiler**

### CODE REVIEW

**Comment by seqan-actions** [comment]
Documentation preview available at https://docs.seqan.de/preview/seqan/seqan3/3292

File: test/api_stability/CMakeLists.txt [hunk 1]
```diff
@@ -28,13 +28,14 @@ macro (seqan3_api_stability_test test_suite git_apply)
         CMAKE_ARGS ${SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS}
                    -DCMAKE_CXX_FLAGS=${SEQAN3_API_STABILITY_CXX_FLAGS}
                    -DCMAKE_CXX_COMPILER_LAUNCHER=${CMAKE_CXX_COMPILER_LAUNCHER}
-                   -DCMAKE_PREFIX_PATH=${CMAKE_CURRENT_LIST_DIR}/../../build_system # use current build_system
+                   -DCMAKE_PREFIX_PATH=${CMAKE_CURRENT_LIST_DIR}/../../cmake # use current cmake
```
Comment (SGSSGene) [inline test/api_stability/CMakeLists.txt:31]: This comment is confusing

File: test/external_project/seqan3_submodule_find_package/CMakeLists.txt [hunk 2]
```diff
@@ -10,7 +10,7 @@ include (../find-package-diagnostics.cmake)
 # ---
 
 # add seqan3 to search path
-list (APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_SOURCE_DIR}/submodules/seqan3/build_system") # change this value
+list (APPEND CMAKE_PREFIX_PATH "${CMAKE_CURRENT_SOURCE_DIR}/submodules/seqan3/cmake") # change this value
```
Comment (SGSSGene) [inline test/external_project/seqan3_submodule_find_package/CMakeLists.txt:13]: Confusing comment, why should this value be changed?

**Review by SGSSGene — CHANGES_REQUESTED** [review]
There are two comments from, I believe, the old times, that have got lost and shouldn't be.

File: test/external_project/seqan3_submodule_find_package/CMakeLists.txt (hunk 2 above)
Comment (eseiler) [reply test/external_project/seqan3_submodule_find_package/CMakeLists.txt]: It might have been a snippet in the tutorial, but it isn't anymore.

**Review by SGSSGene — APPROVED** [review]
very nice!

**Pull request merged**